### PR TITLE
Charlesmchen/fix message details view

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/MessageDetailViewController.swift [hunk 1]
```diff
@@ -238,19 +238,15 @@ class MessageDetailViewController: OWSViewController, MediaGalleryDataSourceDele
...
-                    statusLabel.text = shortStatusMessage
-                    statusLabel.textColor = UIColor.ows_darkGray
-                    statusLabel.font = .ows_dynamicTypeFootnote
-                    statusLabel.adjustsFontSizeToFitWidth = true
-                    statusLabel.sizeToFit()
-                    cell.accessoryView = statusLabel
-                    cell.setContentHuggingLow()
-                    cell.isUserInteractionEnabled = false
-                    groupRows.append(cell)
+                    cellView.accessoryMessage = shortStatusMessage
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/MessageDetailViewController.swift]: Prefer `accessoryMessage` to `accessoryView` for consistent styling.

File: SignalMessaging/Views/ContactCellView.m [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    hStackView.axis = UILayoutConstraintAxisHorizontal;
+    hStackView.spacing = kContactCellAvatarTextMargin;
+    hStackView.distribution = UIStackViewDistributionFill;
+    [self addSubview:hStackView];
+    [hStackView autoVCenterInSuperview];
+    [hStackView autoPinLeadingToSuperviewMargin];
+    [hStackView autoPinTrailingToSuperviewMargin];
+    // Ensure that the cell's contents never overflow the cell bounds.
+    [hStackView autoPinEdgeToSuperviewMargin:ALEdgeTop relation:NSLayoutRelationGreaterThanOrEqual];
+    [hStackView autoPinEdgeToSuperviewMargin:ALEdgeBottom relation:NSLayoutRelationGreaterThanOrEqual];
```
Comment (charlesmchen-signal) [inline SignalMessaging/Views/ContactCellView.m]: Much of this logic is pulled over wholesale from `ContactTableViewCell`, but I took the opportunity to clean up. For example, I use stack views where possible.

File: SignalMessaging/contacts/SystemContactsFetcher.swift [hunk 3]
```diff
@@ -396,8 +396,6 @@ public class SystemContactsFetcher: NSObject {
 
     @objc
     public func fetchCNContact(contactId: String) -> CNContact? {
-        SwiftAssertIsOnMainThread(#function)
```
Comment (charlesmchen-signal) [inline SignalMessaging/contacts/SystemContactsFetcher.swift]: Removed an overzealous assert.

File: SignalMessaging/contacts/OWSContactsManager.m [hunk 4]
```diff
@@ -143,7 +143,6 @@ - (BOOL)supportsContactEditing
 
 - (nullable CNContact *)cnContactWithId:(nullable NSString *)contactId
 {
-    OWSAssert(contactId.length > 0);
```
Comment (charlesmchen-signal) [inline SignalMessaging/contacts/OWSContactsManager.m]: Removed an obsolete assert.

File: SignalMessaging/Views/ContactCellView.m (hunk 2 above)
Comment (michaelkirk-signal) [reply SignalMessaging/Views/ContactCellView.m]: 👍

**Pull request merged**